airflow/serialization/serialized_objects.py

```diff
@@ -839,7 +882,21 @@ def deserialize_operator(cls, encoded_op: Dict[str, Any]) -> Operator:
     @classmethod
     def detect_dependencies(cls, op: Operator) -> Optional['DagDependency']:
         """Detects between DAG dependencies for the operator."""
-        return cls.dependency_detector.detect_task_dependencies(op)
+        dependency_detector = DependencyDetector()
+        custom_dependency_detector = conf.getimport('scheduler', 'dependency_detector')
+        deps = set()
+        if not (custom_dependency_detector is None or type(dependency_detector) is DependencyDetector):
+            warnings.warn(
+                "Use of a custom dependency detector is deprecated. "
+                "Support will be removed in a future release.",
+                DeprecationWarning,
+            )
+            dep = custom_dependency_detector.detect_task_dependencies(op)
+            if dep:
+                deps.add(dep)
+        deps.update(dependency_detector.detect_task_dependencies(op))
```

Comment:
Just checking, we should be calling both the custom and default one?

Reply:
right, we're storing the deps in a set in order to dedupe.  custom "detector" can only return a single value, we just add it to what we get from the current detector.

Reply:
_really_, maybe we don't even need to maintain backcompat because it's just UI, and there's no promise that the UI doesn't change -- e.g. when we removed tree view and replaced with grid.  so since in theory we can freely remove this view and replace it with something else, it seems it would follow that we can also remove the pluggability?